### DISCUSSION TITLE
Refs cannot be passed to render props

### DISCUSSION THREAD

**Discussion by yepitschunked** [reactwg/react-compiler#42] [Something went wrong]
[Playground link](https://playground.react.dev/#N4Igzg9grgTgxgUxALhAMygOzgFwJYSYAEAsgJ4DCEAtgA6EKY4AUwAOsUXABZ4A2AExiMANBwC+yIgAUYEWmACURdpziEwOItTIAlBGiIBeIlDAJ9aADyYo1AEYIYAPmYAGRQG4OPzsJywxMwcRKFEVs4hYaHAPPxCjMGc0aE6llHRiuIZ4QD0kZxeEhwg4kA)

The compiler/lint rule complains with `InvalidReact: Ref values (the `current` property) may not be accessed during render.` if refs are passed as an argument to a [render prop](https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop). 

It's true that the render function could technically misbehave and read the ref during render, but it feels weird to lint for it in the parent component.

**Comment by aeharding** [comment]
Nice, I think this is a simplified version of my repro here: https://github.com/reactwg/react-compiler/discussions/32

**Comment by elnatanitzan** [comment]
This has been reported several times and there is an open issue about:
https://github.com/facebook/react/issues/31637

Currently, it's totally fine to disable eslint rule for this line 🙂

**Reply by aeharding** [reply]
The rule can be disabled, but the compiler still bails.

Thanks for the link!

**Reply by yepitschunked** [reply]
I don't think your link is relevant. That issue is about trying to run compiled JSX through the compiler, which obviously won't work.

**Reply by andreasvirkus** [reply]
There are plans to silence this because of a large number of false-positives, it seems https://github.com/facebook/react/pull/30843
